src/stk500v2.c

```diff
@@ -1620,7 +1628,7 @@ static int stk500hv_initialize(const PROGRAMMER *pgm, const AVRPART *p, enum hvm
       stk500v2_getparm(pgm, PARAM_OSC_CMATCH, &osc_cmatch);
       if(osc_pscale) {
         int prescale = 1;
-        f_get = STK500V2_XTAL / 2;
+        f_get = PDATA(pgm)->xtal / 2;
```

Comment:
Same as above:

Not a fault of this PR, but osc_pscale and osc_cmatch ought to be set to 0 above b/c stk500v2_getparm() might not set them! Ideally, also the return value of stk500v2_getparm() calls ought to be checked to guard against not setting the osc_... variables.